addon/services/construction-project.js

```diff
@@ -93,34 +94,57 @@ export default class BuildingProjectService extends XMLApiService {
     return this.createAndCacheProject(xml);
   }
 
-  async search(query = {}) {
-    // if EPROID is set, ignore other search filters
+  searchProject(query) {
+    return this.search(
+      query,
+      query.EPROID,
+      "getConstructionProject",
+      "constructionprojects",
+      ConstructionProjectsList,
+      "constructionProject",
+      "constructionProjectsList"
```

Comment:
It might be more readable to pass an options object for some of the options, e.g.
```js
return this.search(query, query.EPROID, {
  xmlMethod: "getConstructionProject",
  urlPath: "constructionprojects",
  listModel: ConstructionProjectsList,
  listKey: "constructionProject",
  searchKey: "constructionProjectsList"
});
```

Reply:
The distinction is kind of arbitrary - the choice I made in the example is because I'd like to name "similar-looking" arguments.